service/src/main/java/bio/terra/workspace/service/iam/model/WsmIamRole.java

```diff
@@ -38,6 +42,32 @@ public static WsmIamRole fromSam(String samRole) {
         () -> new RuntimeException("No IamRole enum found corresponding to Sam role " + samRole));
   }
 
+  public static WsmIamRole getHighestRole(UUID workspaceId, List<WsmIamRole> roles) {
+    if (roles.isEmpty()) {
+      throw new InternalServerErrorException(
+          String.format("Workspace %s missing roles", workspaceId.toString()));
+    }
+
+    if (roles.contains(WsmIamRole.APPLICATION)) {
+      Preconditions.checkState(
```

Comment:
Why do we need to require that an application only have APPLICATION role?
I think you get the same effect if you decide that APPLICATION collates highest.